### PR TITLE
Move to a more OO Instance.

### PR DESCRIPTION
**Pull request by mikalstill**
This doesn't completely remove the old way of using the database, but it's a step in that direction without being an impossibly large amount of code to review.

### CODE REVIEW

**Comment by mikalstill** [comment]
Chasing some weirdness in CI.

**Comment by mikalstill** [comment]
The filtering of Instances is causing me some issues...

**Comment by mikalstill** [comment]
This PR fixes #590.

File: shakenfist/virt.py [hunk 1]
```diff
@@ -138,6 +140,14 @@ def __init__(self, static_values):
             LOG.withObj(self).error('Found disk spec empty')
             raise exceptions.InstanceBadDiskSpecification()
 
+    @staticmethod
+    def _db_create_instance(instance_uuid, metadata):
```
Comment (mandoonandy) [inline shakenfist/virt.py:144]: Should this just be in new()?

File: shakenfist/virt.py [hunk 2]
```diff
@@ -793,3 +862,53 @@ def get_console_data(self, length):
...
+                # the result set.
+                if not f(i):
+                    skip = True
+                    break
+
+            if not skip:
+                yield i
+
+
+def placement_filter(node, inst):
```
Comment (mandoonandy) [inline shakenfist/virt.py:889]: This does more than filter?

File: shakenfist/virt.py [hunk 3]
```diff
@@ -793,3 +862,53 @@ def get_console_data(self, length):
             'Client requested %d bytes of console log, returning %d bytes'
             % (length, len(d)))
         return d
+
+
+class Instances(object):
```
Comment (mandoonandy) [inline shakenfist/virt.py:867]: This is not a concept that I find easy to like. If we are going OO maybe this should be a factory class of some sort?

**Review by mandoonandy — COMMENTED** [review]
Looks good overall. I have a little uneasiness about the Instances class, but I am liking the move towards OO'ness making the code easier to read.

**Comment by mikalstill** [comment]
Replaced by #596 because rebase.

**Pull request closed without merging**